crates/provider-sdk/src/rpc_client.rs

```diff
@@ -249,17 +240,31 @@ impl RpcClient {
     /// the appropriate time, an error will be returned.
     #[instrument(level = "debug", skip_all, fields(subject = %subject))]
     pub async fn request(&self, subject: String, payload: Vec<u8>) -> InvocationResult<Vec<u8>> {
-        match maybe_timeout(
-            self.timeout,
-            self.client
-                .request(subject, payload.into())
-                .map_err(|e| InvocationError::from(NetworkError::from(e))),
-        )
-        .await
-        {
+        self.request_timeout(subject, payload, self.timeout).await
+    }
+
+    async fn request_timeout(
+        &self,
+        subject: String,
+        payload: Vec<u8>,
+        timeout: Option<Duration>,
+    ) -> InvocationResult<Vec<u8>> {
+        let timeout = if timeout.is_none() {
+            self.timeout
+        } else {
+            timeout
+        };
+
+        let req = async_nats::Request::new()
+            .payload(payload.into())
+            .timeout(timeout);
```

Comment:
notes: `request` always sets the timeout, which will make the request unaffected by the timeout in the `nats client`